devel/query.lua

```diff
@@ -107,148 +76,326 @@ or specifically what key to print inside a unit.
 
-``-listall``:           Lists both fields and keys, useful if you aren't running a search.
+``-maxdepth <value>``:     Limits the field recursion depth (default: 7)
 
-``-listfields``:        Lists fields. Useful if you aren't running a search.
+``-maxlength <value>``:    Limits the table sizes that will be walked (default: 257)
 
-``-listkeys``:          Lists keys. Useful. Ya, period.
+``-excludetype [a|bfnstu0]``:  Excludes data types: All | Boolean, Function, Number, String, Table, Userdata, nil
 
-``-depth <value>``:          Limits the field recursion depth (default: 10)
+``-excludekind [a|bces]``:     Excludes data types: All | Bit-fields, Class-type, Enum-type, Struct-type
 
-``-keydepth <value>``:       Limits the key recursion depth (default: 4)
+``-noblacklist``:   Disables blacklist filtering.
 
-``-maxtablelength <value>``: Limits the table sizes that will be walked (default: 257)
+``-dumb``:          Disables intelligent checking for recursive data structures(loops)
+                    and increases the -maxdepth to 25 if a value is not already present
 
-``-includeitall``:  Removes blacklist filtering, and disregards readability of output.
+**Command options:**
 
-``-safer``:         Disables walking struct data.
+``-setvalue <value>``: Attempts to set the values of any printed fields.
+                       Supported types: boolean,
 
-                    Unlike native Lua types, struct data can sometimes be misaligned,
-                    which can cause crashes when accessing it. This option may be useful
-                    if you're running an alpha or beta build of DFHack.
+``-oneline``:          Reduces output to one line, except when ``-debugdata`` is used.
 
-``-dumb``:          Disables intelligent checks for things such as reasonable
-recursion depth (i.e. depth maximums are increased, not removed) and also checks
-for recursive data structures (i.e. to avoid walking a child that goes to a
-parent)
+``-disableprint``:     Disables printing. Might be useful if you are debugging this script.
+                       Or to see if a query will crash (faster) but not sure what else you could use it for.
+                       
+``-debug <value>``:    Enables debug log lines equal to or less than the value provided.
 
-**Command options:**
+``-debugdata``:        Enables debugging data. Prints type information under each field.
 
-``-debug <value>``: Enables debug log lines equal to or less than the value
-provided. Some lines are commented out entirely, and you probably won't even use
-this.. but hey, now you know it exists.
+``-help``:             Prints this help information.
 
-``-disableprint``: Disables printing fields and keys. Might be useful if you are
-debugging this script. Or to see if a query will crash (faster) but not sure
-what else you could use it for.
+]====]
 
-``-help``: Prints this help information.
 
-]====]
 
 --[[ Test cases:
```

Comment:
note that you can make actual automated tests out of these. You can add tests to `test/devel/query.lua` and run them with the DFHack `test` command. e.g. `test -d/path/to/your/scripts/repo -tquery`

Reply:
good to know